Workflows/GIT/Clean changes.md

```diff
@@ -0,0 +1,131 @@
+After a while you'll get notifications about comments on your pull requests from your reviewers. Often times, you'll need to change something before the branch is ready.
+
+## Solving change requests
+Change requests should be applied in separate _fixup_ or _squash_ commits. Rebaseing the branch during an ongoing review is not appreciated unless there is a good reason for it, like pulling in some new and necessary changes from `master`, because it makes harder for the reviewers to know what the new changes are and what they already reviewed.
+
+These commits should be merged into `staging` as well when they are done.
+
+## Merging
+
+### Staging
+Once the PR is opened, the branch can be merged into `staging`, unless it contains considerable
+logic in the migrations, in which case the reviewers should prioritize reviewing the migrations
+first, and giving a thumbs up before merging.
+
+We have 2 ways of doing branch merges to `staging`:
+1. Merge & squash
+2. Cherry-picking a commit range
+
+#### Merge squash
+
+```bash
+git switch staging
+git fetch origin staging && git pull --rebase
+git merge --squash {branch-name}
+```
+
+#### Cherry-picking
+
+```bash
+git switch staging
+git fetch origin staging && git pull --rebase
+git cherry-pick -n {BASE-OF-BRANCH}..{branch-name}
+```
+
+_Note_: BASE-OF-BRANCH is one commit prior of the first commit of the branch.
+
+Cherry picking usually produces less merge conflicts once `master` and `staging` diverge.
+
+The commit message should be in the following format.
+```
+Merge-squash {branch-name}
+
+(pull request {pull-request-link})
+[optionally](cherry picked from {commit-sha})
+```
+
+Including the pull request link in the message results with a reference in the pull request feed, which is helpful because we'll know when the branch was deployed to `staging`.
+
+Make sure that everything's ok by running the specs, then push.
+
+#### Keep it clean
+After a while the `staging` branch history can become quite different than `master`, because of the different branch merges. You'll sometimes even find yourself pushing a couple of _fixup_ commits to `staging` if you're fixing something and then applying the changes to the `fix/branch`.
```

Comment:
```suggestion
After a while the `staging` branch history can become quite different than that of the `master` branch, because of the different branch merges. You'll sometimes even find yourself pushing a couple of _fixup_ commits to `staging` if you're fixing something and then applying the changes to the `fix/branch`.
```